lib/io/hardware_adc.cpp

```diff
@@ -54,7 +55,8 @@ std::optional<std::uint16_t> Adc::resetAndRead4(const int file_descriptor)
     return std::nullopt;
   }
   const int raw_voltage = std::atoi(read_buffer);
-  return static_cast<std::uint16_t>(raw_voltage);  // max value is 2^12-1 = 4095
+  // convert raw voltage to voltage between [0, 1.8] in V
```

Comment:
Do we not check the values of the raw voltage against the maximum allowed value?

Reply:
Also not sure if I'm missing something here but if the value is between 0 and 1.8, why do we return `uint16_t`?

Reply:
this is truly one of the casts of all time :00, will change asap - wrote this after a tiring testing sesh